src/main/java/com/users/service/UserService.java

```diff
@@ -0,0 +1,78 @@
+package com.users.service;
+
+import com.users.dto.UserDto;
+import com.users.entities.User;
+import com.users.exceptions.EntityAlreadyExistsException;
+import com.users.exceptions.EntityNotFoundException;
+import com.users.mappers.UserMapper;
+import com.users.repository.UserRepository;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.Period;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Transactional
+@AllArgsConstructor
+@Service
+@Slf4j
+public class UserService {
+    private UserRepository userRepository;
+    private UserMapper userMapper;
+
+    public UserDto create(User user) {
+        log.info("User to be created {}", user);
+        List<User> allByEmail = userRepository.findAllByEmail(user.getEmail());
+        if (!allByEmail.isEmpty()) {
+            throw new EntityAlreadyExistsException(String.format("User with email %s already exists", user.getEmail()));
+        }
+        validateDateOfBirth(user.getDateOfBirth());
+        User savedUser = userRepository.save(user);
+        return userMapper.toDto(savedUser);
+    }
+
+    public UserDto updateUser(Long id, UserDto userDto) {
+        log.info("Update user with id {}", id);
+        User user = userRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(String.format("User with id %s doesn't exist", id)));
+        validateDateOfBirth(user.getDateOfBirth());
```

Comment:
la fel ca mai sus, validarea ar trb sa fie dupa log, acolo unde se poate, plus ca tu nu trebuie sa validezi ce ai in baza de date (user), tu trebuie sa validezi ce vine pe request

Reply:
deci eu trebuie sa fac un check pe userDto, cel pe care il am in parametrii mei, pentru ca daca fac pe user fac degeaba pentru ca dupa oricum fac update pe el. Oricum era degeaba pe user ca are metoda deja facute pe create.

Reply:
tu trebuie sa faci validare pe ceva ce vrei sa salvezi, nu pe ce este deja salvat, user e deja in baza de date, ca e sau nu mai mare de 18 ani nu ne mai intereseaza aici, aici ne intereseaza ce urmeaza sa salvam

Reply:
adica pe parametrul userDto trebuie sa fac pentru ca pe el urmeaza sa l salvez, probabil o sa vina ca (@PathVariable Long id, @RequestBody UserDto) in EdnPoint si daca eu pun o data aiurea la userDto o sa mi dea eroarea respectiva

Reply:
nu vorim aici de controller, vorbim strict de metoda asta  validateDateOfBirth(user.getDateOfBirth()); nu trebuie aplicata pe user.getDateOfBirth, trb aplicata pe userDto.getDateOfBirth

Reply:
da bossule, asta am zis si eu de 2 ori mai sus, dar nu vrei tu sa mi dai dreptate.  Am schimbat din prima in Intellij
public UserDto updateUser(Long id, UserDto userDto) {
        log.info("Update user with id {}", id);
        validateDateOfBirth(userDto.getDateOfBirth());
        User user = userRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException(String.format("User with id %s doesn't exist", id)));
        userMapper.updateEntity(user, userDto);
        User savedUser = userRepository.save(user);
        return userMapper.toDto(savedUser);
    }

Reply:
pai iti explicam de ce trebuie sa faci asta :)) si nu mi aparuse ca ai schimbat :))